axelor-stock/src/main/java/com/axelor/apps/stock/service/PickedProductsServiceImpl.java

```diff
@@ -0,0 +1,288 @@
+import com.axelor.apps.stock.db.repo.StockMoveRepository;
+import com.axelor.apps.stock.db.repo.StoredProductsRepository;
+import com.axelor.apps.stock.exception.StockExceptionMessage;
+import com.axelor.db.JPA;
+import com.axelor.i18n.I18n;
+import com.axelor.inject.Beans;
+import com.google.inject.Inject;
+import com.google.inject.persist.Transactional;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+
+public class PickedProductsServiceImpl implements PickedProductsService {
+  protected PickedProductsRepository pickedProductsRepository;
+  protected StoredProductsRepository storedProductsRepository;
+  protected MassStockMoveRepository massStockMoveRepository;
+  protected StockMoveRepository stockMoveRepository;
+  protected StockMoveLineRepository stockMoveLineRepository;
+  protected StockLocationLineRepository stockLocationLineRepository;
+  protected StockMoveService stockMoveService;
+  protected StockMoveLineService stockMoveLineService;
+
+  @Inject
+  public PickedProductsServiceImpl(
+      PickedProductsRepository pickedProductsRepository,
+      StoredProductsRepository storedProductsRepository,
+      MassStockMoveRepository massStockMoveRepository,
+      StockMoveRepository stockMoveRepository,
+      StockMoveLineRepository stockMoveLineRepository,
+      StockLocationLineRepository stockLocationLineRepository,
+      StockMoveService stockMoveService,
+      StockMoveLineService stockMoveLineService) {
+    this.pickedProductsRepository = pickedProductsRepository;
+    this.storedProductsRepository = storedProductsRepository;
+    this.massStockMoveRepository = massStockMoveRepository;
+    this.stockMoveRepository = stockMoveRepository;
+    this.stockMoveLineRepository = stockMoveLineRepository;
+    this.stockLocationLineRepository = stockLocationLineRepository;
+    this.stockMoveService = stockMoveService;
+    this.stockMoveLineService = stockMoveLineService;
+  }
+
+  @Override
+  @Transactional(rollbackOn = Exception.class)
+  public PickedProducts createPickedProduct(
+      MassStockMove massStockMove,
+      Product pickedProduct,
+      TrackingNumber trackingNumber,
+      StockLocation fromStockLocation,
+      BigDecimal currentQty,
+      BigDecimal pickedQty,
+      StockMoveLine stockMoveLine) {
+
+    PickedProducts pickedProducts = new PickedProducts();
+    pickedProducts.setPickedProduct(pickedProduct);
+    pickedProducts.setTrackingNumber(trackingNumber);
+    pickedProducts.setFromStockLocation(fromStockLocation);
+    pickedProducts.setCurrentQty(currentQty);
+    pickedProducts.setUnit(pickedProduct.getUnit());
+    pickedProducts.setPickedQty(pickedQty);
+    pickedProducts.setStockMoveLine(stockMoveLine);
+    pickedProducts.setMassStockMove(massStockMove);
+    pickedProductsRepository.save(pickedProducts);
+    return pickedProducts;
+  }
+
+  @Override
+  @Transactional(rollbackOn = Exception.class)
+  public void createStockMoveAndStockMoveLine(
```

Comment:
Please refactor this method (refactor should be similar as StoredProductServiceImpl)